src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpMessageHandlerIntegration.cs

```diff
@@ -0,0 +1,118 @@
+        internal const string ServiceName = "http-client";
+
+        // internal readonly  string Name = nameof(HttpMessageHandlerIntegration).Substring(0, )
+
+        /// <summary>
+        /// Instrumentation wrapper for <see cref="HttpMessageHandler.SendAsync"/>.
+        /// </summary>
+        /// <param name="handler">The <see cref="HttpMessageHandler"/> instance to instrument.</param>
+        /// <param name="request">The <see cref="HttpRequestMessage"/> that represents the current HTTP request.</param>
+        /// <param name="cancellationTokenSource">The <see cref="CancellationTokenSource"/> that can be used to cancel this <c>async</c> operation.</param>
+        /// <returns>Returns the value returned by the inner method call.</returns>
+        [InterceptMethod(
+            TargetAssembly = "System.Net.Http",
+            TargetType = "System.Net.Http.HttpMessageHandler")]
+        [InterceptMethod(
+            TargetAssembly = "System.Net.Http",
+            TargetType = "System.Net.Http.HttpClientHandler")]
+        public static object SendAsync(
+            object handler,
+            object request,
+            object cancellationTokenSource)
+        {
+            // HttpMessageHandler
+            // Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            var tokenSource = cancellationTokenSource as CancellationTokenSource;
+            var cancellationToken = tokenSource?.Token ?? CancellationToken.None;
+
+            return SendAsyncInternal(
+                (HttpMessageHandler)handler,
+                (HttpRequestMessage)request,
+                cancellationToken);
+        }
+
+        private static async Task<HttpResponseMessage> SendAsyncInternal(
+            HttpMessageHandler handler,
+            HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            var executeAsync = DynamicMethodBuilder<Func<HttpMessageHandler, HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>>>
+               .GetOrCreateMethodCallDelegate(
+                    handler.GetType(),
+                    nameof(SendAsync));
+
+            if (!IsTracingEnabled(request))
+            {
+                return await executeAsync(handler, request, cancellationToken).ConfigureAwait(false);
+            }
+
+            using (var scope = CreateScope(handler, request))
+            {
+                try
+                {
+                    // add distributed tracing headers
+                    request.Headers.Inject(scope.Span.Context);
+
+                    HttpResponseMessage response = await executeAsync(handler, request, cancellationToken).ConfigureAwait(false);
+                    scope.Span.SetTag(Tags.HttpStatusCode, ((int)response.StatusCode).ToString());
+                    return response;
+                }
+                catch (Exception ex)
```

Comment:
👍 I'm also in favor or using exception filtering. Although I'd say that to be consistent this should probably be the topic of another PR since we do it in many places.

Reply:
Ok, then. I created a Trello ticket so we can follow up.